ibis/expr/format.py

```diff
@@ -145,43 +146,61 @@ def inline_args(fields, prefer_positional=False):
     return ", ".join(f"{k}={v}" for k, v in fields.items())
 
 
+def find_defining_frame(node):
+    for frame_info in inspect.stack()[::-1]:
+        for var in frame_info.frame.f_locals.values():
+            if isinstance(var, ir.Expr) and var.op() == node:
```

Comment:
The equality check feels like it will cause a bunch of headaches when debugging very large expressions.

Just want to call that out as a potential pitfall here.

Reply:
It is an advanced opt-in configuration mostly going to be used by us. If it turns out to be stable we can build more advanced UX features assisting the users.